src/server/scripts/Kalimdor/CavernsOfTime/WellOfEternity/instance_well_of_eternity.cpp

```diff
@@ -45,7 +53,104 @@ class instance_well_of_eternity : public InstanceMapScript
             SetBossNumber(EncounterCount);
             LoadObjectData(creatureData, nullptr);
             LoadDungeonEncounterData(encounters);
+
+            PerotharnIntroState = NOT_STARTED;
+            IllidanEscortState = NOT_STARTED;
+            IllidanIntroState = NOT_STARTED;
+            IllidanFirstCrystalState = NOT_STARTED;
+            _crystalOneCreaturesAlive = 0;
+            //_crystalTwoCreaturesAlive = 0;
+            //_crystalThreeCreaturesAlive = 0;
+        }
+
+        void OnPlayerEnter(Player* /*player*/) override
+        {
+            if (GetData(DATA_ILLIDAN_START_ESCORT) == DONE)
+                DoCastSpellOnPlayers(SPELL_SHADOWCLOAK_PLAYER);
```

Comment:
Why recast the spell on all players when any player enters the instance?

Reply:
if a player dies while is escorting illidan or exits the instance the aura is removed, and when enters again (the graveyard is outside the dungeon) the spell is recasted

Reply:
Doesn't answer my question - you have 4 players inside, why are you casting this on 5 players (4 inside + 1 entering) when 5th one enters?